pkg/nuclide-diagnostics-store/lib/LinterAdapter.js

```diff
@@ -149,7 +149,16 @@ export class LinterAdapter {
 
   async _runLint(editor: TextEditor): Promise<void> {
     if (this._enabled) {
-      const result = await this._requestSerializer.run(this._provider.lint(editor));
+      const maybe = this._provider.lint(editor);
+      if (!maybe) {
+        return;
+      }
+
+      const result = await this._requestSerializer.run(maybe);
+      if (!result) {
```

Comment:
`RequestSerializer::run` never returns `null` or `undefined`. If the underlying promise resolves to `null` it will return an object with a `null` `result` field.

Instead you'll need to check `linterMessages` (below).

This change should also change the types in `nuclide-diagnostics-common/lib/main.js` to reflect the actual `lint` return type. Then Flow should guide you through the necessary changes.